include/flashinfer/norm.cuh

```diff
@@ -142,9 +142,15 @@ __global__ void FusedAddRMSNormKernel(T* __restrict__ input, T* __restrict__ res
       float x = float(input_vec[j]);
       x += float(residual_vec[j]);
       sum_sq += x * x;
-      residual_vec[j] = (T)x;
+      if constexpr (sizeof(T) == 2) {
+        input_vec[j] = *((T *)((uint16_t *)(&x)) + 1);
+        residual_vec[j] = *(T *)((uint16_t *)(&x));
+      } else {
+        residual_vec[j] = (T)x;
+      }
     }
     if ((i * num_threads + thread_id) * VEC_SIZE < d) {
+      input_vec.store(input + bx * d + i * num_threads * VEC_SIZE + thread_id * VEC_SIZE);
```

Comment:
Can we use shared memory instead of writing to and reading from global memory inside kernels?
Same for `residual_vec`.

Reply:
> Can we use shared memory instead of writing to and reading from global memory inside kernels? Same for `residual_vec`.

Thank you for your response, I will attempt to modify it to use shared memory.